src/Network/Network/FirewallPolicy/PolicySettings/LogScrubbing/LogScrubbingRule/NewAzureApplicationGatewayFirewallPolicyLogScrubbingRuleCommand.cs

```diff
@@ -0,0 +1,74 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApplicationGatewayFirewallPolicyLogScrubbingRule"), OutputType(typeof(PSApplicationGatewayFirewallPolicyLogScrubbingRule))]
+    public class NewAzureApplicationGatewayFirewallPolicyLogScrubbingRuleCommand : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "State of the log scrubbing rule. Default value is Enabled")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("Enabled", "Disabled", IgnoreCase = true)]
+        public string State { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The variable to be scrubbed from the logs.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("RequestHeaderNames", "RequestCookieNames", "RequestArgNames", "RequestPostArgNames", "RequestJSONArgNames", "RequestIPAddress", IgnoreCase = true)]
+        public string MatchVariable { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "When matchVariable is a collection, operate on the selector to specify which elements in the collection this rule applies to.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("Equals", "EqualsAny", IgnoreCase = true)]
+        public string SelectorMatchOperator { get; set; }
+
+        [Parameter(
+         Mandatory = false,
+         HelpMessage = "When matchVariable is a collection, operator used to specify which elements in the collection this rule applies to.")]
+        [ValidateNotNullOrEmpty]
```

Comment:
Selector can be null or empty if operator is EqualsAny. can you remove this validation?

Reply:
I think in that case user can just not entering the field Selector rather than putting -Selector null as the field is not mandatory.

Reply:
Do we have a test with EqualsAny operator to ensure this will not throw a validation error?

Reply:
Yes, there is a test case that I wrote where SelectorMatchOperator is EqualsAny and there is no field Selector 
$logScrubbingRule2 = New-AzApplicationGatewayFirewallPolicyLogScrubbingRule -State Enabled -MatchVariable RequestIPAddress -SelectorMatchOperator EqualsAny

Reply:
@sindhualuguvelli1 do you still think we need this removed?

Reply:
I think it's okay to keep it.